### PR TITLE
[SRVC-17749] Improve Gem Version & Info Merge to combine based on Digest

### PR DESCRIPTION
**Pull request by brand-it**
#### Error or Issue

```
bundle update ddtrace
Fetching gem metadata from http://localhost:9292/.....
Resolving dependencies...
Could not find compatible versions

Because ddtrace >= 1.12.0 depends on libddwaf ~> 1.9.0.0.0
  and libddwaf ~> 1.9.0.0.0 could not be found in rubygems repository http://localhost:9292/ or installed locally for any resolution platforms (ruby),
  ddtrace >= 1.12.0 cannot be used.
So, because Gemfile depends on ddtrace >= 1.12,
  version solving has failed.

The source contains the following gems matching 'libddwaf (~> 1.9.0.0.0)':
  * libddwaf-1.9.0.0.0-arm64-darwin
  * libddwaf-1.9.0.0.0-x86_64-darwin

```

#### Testing

```
rakeup
```

Gemfile
```ruby
source "http://localhost:9292"


group :development do
  gem 'byebug'
  gem 'rubocop', '~> 1.31.0'
  gem 'rubocop-minitest', '~> 0.20.1'
  gem 'rubocop-rake', '~> 0.6.0'
end
group :test do
  gem 'minitest'
  gem 'minitest-reporters'
  gem 'rack-test'
  gem 'rake'

  gem 'capybara-mechanize'
  # Pin capybara to 3.36.x or earlier as 3.37.x does not work with Ruby 2.7/3.0/3.1
  # see https://github.com/teamcapybara/capybara/pull/2546
  gem 'capybara', '< 3.37.0'

  gem 'webmock'

  # Used only in test/requests/atom_feed_test.rb
  gem "rss", require: false

  gem "simplecov"
end

```

```
bundle install
```

🍺

### CODE REVIEW

**Review by jcraigk — APPROVED** [review]

File: lib/geminabox/gem_versions_merge.rb [hunk 1]
```diff
@@ -1,54 +1,83 @@
...
+    end
+
+    def preamble_time
+      Time.parse(preamble[0].split[1])
+    end
+
+    def datadir
+      Geminabox.data
+    end
+
```
Comment (brand-it) [inline lib/geminabox/gem_versions_merge.rb:72]:
```suggestion
```

Unused Code

**Review by commitshappen — APPROVED** [review]

**Comment by wmjeffrey** [comment]
Screen Shot 2023-06-09 at 2 06 09 PM

**Comment by wmjeffrey** [comment]
Screen Shot 2023-06-09 at 2 06 19 PM

**Comment by wmjeffrey** [comment]
QA approved! Pointed Core at local geminabox server and bundle install ran successfully

**Review by wmjeffrey — APPROVED** [review]

**Pull request merged**